js/mod.ts

```diff
@@ -217,3 +245,57 @@ function checkCompilerOptions(
     );
   }
 }
+
+interface SerializedImportMap {
+  url: string;
+  value: string;
+}
+
+async function fetchImportMap(
+  urlOrString: URL | string | undefined,
+): Promise<SerializedImportMap | undefined> {
+  if (!urlOrString) return undefined;
+  const url = urlOrString instanceof URL
+    ? urlOrString
+    : toFileUrl(resolve(urlOrString));
+  const fetchImportMapResponse = await fetch(url);
+  const body = await fetchImportMapResponse.json();
+  return {
+    url: url.toString(),
+    value: JSON.stringify({
+      imports: body.imports,
+      scopes: body.scopes,
+    }),
+  };
+}
```

Comment:
I am not particularly happy with this bit of code, but it made things easier as debugging things on the Rust side is a real hassle.

I can change this by forwarding `importMap` instead, then using either `ImportMap::new` or `import_map::parse_from_value` depending on how the import map is provided.

Reply:
> I can change this by forwarding importMap to Rust instead, then using either import_map::ImportMap::new or import_map::parse_from_value depending on how the import map is provided.

I did it, but I kept the JSON serialization.